### ISSUE TITLE
sails-mongo@0.9.x had a problem in adapter.js

### ISSUE THREAD

**Issue by xiwan** [balderdashy/sails-mongo#175]
My project use the **sails-mongo@0.9.x**

```javascript
    registerCollection: function(collection, cb) {
      var self = this;

      // Load the url connection parameters if set
      collection.config = utils.parseUrl(collection.config);

      // If the configuration in this collection corresponds
      // with a known database, reuse it the connection(s) to that db
      dbs[collection.identity] = _.find(dbs, function(db) {
        return collection.database === db.database;
      });

      // Otherwise initialize for the first time
      if (!dbs[collection.identity]) {
        dbs[collection.identity] = collection;
      }

      // Holds the Schema
      dbs[collection.identity].schema = {};
      schemaStash[collection.identity] = collection.definition;

      return cb();
    },
```

in the middle part,  the code will reuse the connection if a known database exist. BUT what if both of **collection.database** and **db.database** are **undefined**? After my test, it will always return the **first connection of dbs**. No matter how hard I try to overwrite 'default' adapter in model, the first connection of dbs (**default adapter**) will be choosen.

Please correct me if anything mis-understood.

Thanks in advance.

**Comment by sailsbot** [comment]
Thanks for posting, @xiwan.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
